### PR TITLE
[Alpha]Implements Get Total Number

### PR DESCRIPTION
**Pull request by Lorraine-Lyu**

### CODE REVIEW

File: src/main/webapp/js/datahandlers/listdatahandler.js [hunk 1]
```diff
@@ -68,6 +68,7 @@ class ListDataHandler {
           await fetch(this.getPath_(batchIndex, itemsPerBatch, lastIndex));
       let listJson = await responseList.json();
       if (type == COLLEGES) {
+        console.log(listJson);
```
Comment (MikolJiron) [inline src/main/webapp/js/datahandlers/listdatahandler.js]: Might want to remove this for devel if it was just used for debugging.

**Review by MikolJiron — CHANGES_REQUESTED** [review]

File: src/main/webapp/js/datahandlers/listdatahandler.js (hunk 1 above)
Comment (Lorraine-Lyu) [reply src/main/webapp/js/datahandlers/listdatahandler.js]: Yep just deleted it.

**Review by MikolJiron — APPROVED** [review]

File: src/main/webapp/js/datahandlers/scholarshiplistdatahandler.js [hunk 2]
```diff
@@ -38,8 +39,9 @@ class ScholarshipListDataHandler extends ListDataHandler {
    * The total number of scholarship stored in backend.
    */
   async getTotalNumber() {
-    // TODO: fetch from backend.
-    return 35;  // Presetting page length not supported yet.
+    const data = await fetch(DATA_SIZE_ENDPOINT);
+    const number = await data.text();
+    return parseInt(number);
```
Comment (CarlosM10011) [inline src/main/webapp/js/datahandlers/scholarshiplistdatahandler.js]: add base to parseInt. Also, how is NaN handled?

`return parseInt(number, 10);`

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]
Please pull then git rebase devel then git push -f.

File: src/main/webapp/js/datahandlers/scholarshiplistdatahandler.js (hunk 2 above)
Comment (Lorraine-Lyu) [reply src/main/webapp/js/datahandlers/scholarshiplistdatahandler.js]: good catch! thanks!

**Review by CarlosM10011 — APPROVED** [review]

File: src/main/java/com/google/step/finscholar/servlets/DataSizeServlet.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+@WebServlet("/data-size")
+public class DataSizeServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Optional<Firestore> database = Optional.empty();
+    
+    database = Optional.ofNullable(FirestoreClient.getFirestore(FirebaseAppManager.getApp().get()));
+    if (database.isPresent()) {
```
Comment (tinagrasso) [inline src/main/java/com/google/step/finscholar/servlets/DataSizeServlet.java:47]: What happens if no db is present?

File: src/main/webapp/js/datahandlers/scholarshiplistdatahandler.js [hunk 4]
```diff
@@ -38,8 +39,14 @@ class ScholarshipListDataHandler extends ListDataHandler {
...
    */
   async getTotalNumber() {
-    // TODO: fetch from backend.
-    return 35;  // Presetting page length not supported yet.
+    try {
+      const data = await fetch(DATA_SIZE_ENDPOINT);
+      const numberString = await data.text();
+      return parseInt(numberString, 10);
+    } catch(e) {
+      console.log(e);
```
Comment (tinagrasso) [inline src/main/webapp/js/datahandlers/scholarshiplistdatahandler.js]: No need for the console.log. Throwing the error will log it to the console.

File: src/main/webapp/js/datahandlers/scholarshiplistdatahandler.js [hunk 5]
```diff
@@ -38,8 +39,14 @@ class ScholarshipListDataHandler extends ListDataHandler {
    * The total number of scholarship stored in backend.
    */
   async getTotalNumber() {
-    // TODO: fetch from backend.
-    return 35;  // Presetting page length not supported yet.
+    try {
+      const data = await fetch(DATA_SIZE_ENDPOINT);
+      const numberString = await data.text();
+      return parseInt(numberString, 10);
```
Comment (tinagrasso) [inline src/main/webapp/js/datahandlers/scholarshiplistdatahandler.js]: For constants like you have, put a comment.

```
parseInt(numberString, /** radix= */ 10);
```

**Review by tinagrasso — CHANGES_REQUESTED** [review]
Please provide testing steps you took to verify that this change works as intended.

File: src/main/java/com/google/step/finscholar/servlets/DataSizeServlet.java (hunk 3 above)
Comment (Lorraine-Lyu) [reply src/main/java/com/google/step/finscholar/servlets/DataSizeServlet.java:47]: Oh actually if the db is not present, `Firestore.getFirestore` will throw error. nvm I just added the send error code.

**Comment by Lorraine-Lyu** [comment]
About testing:
The total number of scholarships(45) matches with the actual number of documents in our db.
I'm not sure how to verify the college number, but I consistently get 103 for college data, and our webpage doesn't throw error with 103 being our total number.

**Review by tinagrasso — APPROVED** [review]

**Pull request merged**